### DISCUSSION TITLE
Oldest supported CMake version

### DISCUSSION THREAD

**Discussion by ntadej** [maplibre/maplibre-native#1019] [Strategy]
What is the oldest supported CMake version that we want to support? Currently it is 3.10 to support some older platforms, but it does not support `OBJECT` libraries.

We may want to bump the minimum version to be able to simplify the configuration. Something like 3.19 or 3.21 (inspired by Qt) makes sense.

Here we are mainly constrained by Linux distributions. Tagging @rinigus, @acalcutt, @jwinarske for feedback as known Linux users.

Tagging also the maintainers @ovivoda and @louwers.

**Comment by rinigus** [comment]
Thank you very much for your ping! SailfishOS ships currently with cmake `3.19`. So, going above that will be a major problem for me.

I also support older versions of SailfishOS with the oldest one having cmake `3.11`. Ideally, if possible, would be good to keep it.

When were OBJECT libraries introduced? Please keep it is as low as possible - then I don't have to cut off older devices users from the maps app.

**Reply by ntadej** [reply]
3.12 was the version. For now we have checks in the code and we can keep them for a while longer.

**Reply by rinigus** [reply]
I checked how cmake was updated and, on SailfishOS, it was a jump from 3.11 to 3.19. So, if we are going to bump it, it maybe 3.19 as well. 

In some respect, please judge on how much effort is it to keep support for older cmake. So, if it is getting complicated, go for 3.19 support. But please don't go higher :)

**Comment by jwinarske** [comment]
@ntadej 
CMake versions in my current environments: 
* Ubuntu 22 - 3.22.1
* Yocto (Dunfell) - 3.22.3
* Yocto (Kirkstone) - 3.22.3
* Fedora 37 - 3.26.1
* Clion 2023.1.1 - 3.25.2

My vote is <= 3.22.3
